### DISCUSSION TITLE
Debugging pdm with vscode

### DISCUSSION THREAD

**Discussion by Airwave1981** [Second-Hand-Friends/kleinanzeigen-bot#178] [General]
Hi everyone,

I'm not a python developer, but quite good in other languages as C#.
But I can also write python if I like.

I want to help here a bit more, but I'm struggeling to get the project running in visual studio code debugger.
Sth. with this pdm. I guess.

Maybe anyone could write some notes how to get it debugged in vscode like:

1. Open project folder via vscode
2. Open Run and debug...

Thanks  in advance.

**Comment by Airwave1981** [comment]
Think I need to configure the launch.json properly.
Currently I have:

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Python: Profil kleinanzeigen",
            "type": "python",
            "request": "launch",
            "program": "${file}",
            "console": "integratedTerminal",
            "args": ["--config=E:\\kleinanzeigen-bot\\myadds.yaml"]
        }
    ]
}
```

If I then run on "__main__.py" I get
ModuleNotFoundError: No module named 'kleinanzeigen_bot'


Hope anyone can help.

Thanks

**Comment by Airwave1981** [comment]
Or maybe you could tell me how you develop? Different environment?

Thanks

**Comment by sebthom** [comment]
I am using Eclipse with PyDev and debugging works just fine.

![image](https://github.com/Second-Hand-Friends/kleinanzeigen-bot/assets/426959/b5c35843-4e92-48d1-934d-33c77c8a8842)
![image](https://github.com/Second-Hand-Friends/kleinanzeigen-bot/assets/426959/bb6ddae6-5722-4e1b-bbc0-4b30a723dcf7)
![image](https://github.com/Second-Hand-Friends/kleinanzeigen-bot/assets/426959/15a5a949-2edd-4e1a-879f-5492e84ab22d)

You can also use the command line debugger via `pdm debug publish --config ...`, for more infos how to use it check here https://realpython.com/python-debugging-pdb/ and here https://docs.python.org/3/library/pdb.html

**Reply by panzli** [reply]
Thanks, but is it possible, if I don't know the matcher of a button which I want to click, that I run the script until this point and manually can try to click the button by "trial and error" finding the right XPath?

**Reply by sebthom** [reply]
I haven't used it but maybe an extension like this https://chromewebstore.google.com/detail/selectorshub-xpath-helper/ndgimibanhlabgdgjcpbbndiehljcpfh for your browser helps to find the right XPath expression.
